### PR TITLE
TPC QC: make IDC plots (v. 1)

### PR DESCRIPTION
**Pull request by tklemenz**

### CODE REVIEW

**Comment by tklemenz** [comment]
This is the first version of the IDCs visualization for QC.

Fourier coefficients are not yet included (timestamp -1 for getting the data from the CCDB is not working for them for now) and there is a fit missing to the radial IDC0 distribution. 
This will come in a later update.

**Comment by tklemenz** [comment]
Clang format gives an error at the moment, that is why the check is failing.

`$ clang-format-6.0 -style=file -i IDCs.cxx`
`YAML:52:19: error: unknown key 'DeriveLineEnding'`
`DeriveLineEnding: true`
`Error reading /home/tklemenz/AliSoftware/QualityControl/.clang-format: Invalid argument`

I will investigate.

**Comment by tklemenz** [comment]
The issue with clang-format is fixed and the Fourier Coefficients are now included.

File: Detectors/TPC/calibration/include/TPCCalibration/IDCCCDBHelper.h [hunk 1]
```diff
@@ -136,6 +142,16 @@ class IDCCCDBHelper
...
+  TCanvas* drawIDCZeroCanvas(TCanvas* outputCanvas, std::string_view type, int nbins1D, float xMin1D, float xMax1D, int integrationInterval = -1) const { return drawIDCZeroCanvasHelper(outputCanvas, type, nbins1D, xMin1D, xMax1D, integrationInterval); }
+
+  TCanvas* drawIDCZeroRadialProfile(TCanvas* outputCanvas, int nbinsY, float yMin, float yMax) const { return drawIDCZeroRadialProfileHelper(outputCanvas, nbinsY, yMin, yMax); }
+
+  TCanvas* drawIDCZeroStackCanvas(TCanvas* outputCanvas, Side side, std::string_view type, int nbins1D, float xMin1D, float xMax1D, int integrationInterval = -1) const { return drawIDCZeroStackCanvasHelper(outputCanvas, side, type, nbins1D, xMin1D, xMax1D, integrationInterval); }
+
+  TCanvas* drawIDCOneCanvas(TCanvas* outputCanvas, int nbins1D, float xMin1D, float xMax1D, int integrationIntervals = -1) { return drawIDCOneCanvasHelper(outputCanvas, nbins1D, xMin1D, xMax1D, integrationIntervals); }
+
+  TCanvas* drawFourierCoeff(TCanvas* outputCanvas, Side side, int nbins1D, float xMin1D, float xMax1D) const { return drawFourierCoeffHelper(outputCanvas, side, nbins1D, xMin1D, xMax1D); }
+
```
Comment (wiechula) [inline Detectors/TPC/calibration/include/TPCCalibration/IDCCCDBHelper.h]: Why do you need these functions twice, once in the private and then wrapped here in the public part? Can't you simply move the definitions from the private part here?

File: Detectors/TPC/calibration/include/TPCCalibration/IDCDrawHelper.h [hunk 2]
```diff
@@ -56,6 +58,10 @@ class IDCDrawHelper
   /// \param minZ min z value for drawing (if minZ > maxZ automatic z axis)
   /// \param maxZ max z value for drawing (if minZ > maxZ automatic z axis)
   static void drawSide(const IDCDraw& idc, const o2::tpc::Side side, const std::string zAxisTitle, const std::string filename, const float minZ = 0, const float maxZ = -1);
+  static TH2Poly* drawSide(const IDCDraw& idc, const o2::tpc::Side side, const std::string zAxisTitle);
```
Comment (wiechula) [inline Detectors/TPC/calibration/include/TPCCalibration/IDCDrawHelper.h:61]: If you plan to use `TH2Poly` in QC one has to be very careful, I think. These will be huge ...

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx [hunk 3]
```diff
@@ -99,6 +104,284 @@ unsigned int o2::tpc::IDCCCDBHelper<DataT>::getUngroupedIndexGlobal(const unsign
...
+    canv = new TCanvas(fmt::format("c_sides_{}", type.data()).data(), fmt::format("{}", type.data()).data(), 1000, 1000);
+  }
+
+  IDCDrawHelper::IDCDraw drawFun;
+  drawFun.mIDCFunc = idcFunc;
+  const std::string zAxisTitle = IDCDrawHelper::getZAxisTitle(IDCType::IDCZero);
+  TH2Poly* hAside2D = nullptr;
+  TH2Poly* hCside2D = nullptr;
+  hAside2D = IDCDrawHelper::drawSide(drawFun, Side::A, zAxisTitle);
+  hCside2D = IDCDrawHelper::drawSide(drawFun, Side::C, zAxisTitle);
```
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCCCDBHelper.cxx]:
```suggestion
  auto hAside2D = IDCDrawHelper::drawSide(drawFun, Side::A, zAxisTitle);
  auto hCside2D = IDCDrawHelper::drawSide(drawFun, Side::C, zAxisTitle);
```

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx [hunk 4]
```diff
@@ -99,6 +104,284 @@ unsigned int o2::tpc::IDCCCDBHelper<DataT>::getUngroupedIndexGlobal(const unsign
...
+  hAside2D->SetTitle(fmt::format("{} (A-Side)", type.data()).data());
+  hCside2D->SetTitle(fmt::format("{} (C-Side)", type.data()).data());
+
+  TH1F* hAside1D = nullptr;
+  TH1F* hCside1D = nullptr;
+  hAside1D = IDCDrawHelper::drawSide(drawFun, fmt::format("{}", type.data()).data(), Side::A, nbins1D, xMin1D, xMax1D);
+  hCside1D = IDCDrawHelper::drawSide(drawFun, fmt::format("{}", type.data()).data(), Side::C, nbins1D, xMin1D, xMax1D);
```
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCCCDBHelper.cxx]:
```suggestion
  auto hAside1D = IDCDrawHelper::drawSide(drawFun, fmt::format("{}", type.data()).data(), Side::A, nbins1D, xMin1D, xMax1D);
  auto hCside1D = IDCDrawHelper::drawSide(drawFun, fmt::format("{}", type.data()).data(), Side::C, nbins1D, xMin1D, xMax1D);
```

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx [hunk 5]
```diff
@@ -99,6 +104,284 @@ unsigned int o2::tpc::IDCCCDBHelper<DataT>::getUngroupedIndexGlobal(const unsign
...
+  hCside2D->SetTitleOffset(1.05, "XY");
+  hCside2D->SetTitleSize(0.05, "XY");
+  o2::tpc::painter::drawSectorsXY(Side::C);
+
+  canv->cd(3);
+  hAside1D->Draw();
+
+  canv->cd(4);
+  hCside1D->Draw();
+
```
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCCCDBHelper.cxx:174]:
```suggestion
  hAside2D->SetBit(TObject::kCanDelete);
  hCside2D->SetBit(TObject::kCanDelete);
  hAside1D->SetBit(TObject::kCanDelete);
  hCside1D->SetBit(TObject::kCanDelete);

```

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx [hunk 6]
```diff
@@ -99,6 +104,284 @@ unsigned int o2::tpc::IDCCCDBHelper<DataT>::getUngroupedIndexGlobal(const unsign
...
+    canv = outputCanvas;
+  } else {
+    canv = new TCanvas("c_sides_IDC0_radialProfile", "IDC0", 1000, 1000);
+  }
+
+  IDCDrawHelper::IDCDraw drawFun;
+  drawFun.mIDCFunc = idcFunc;
+  const std::string zAxisTitle = IDCDrawHelper::getZAxisTitle(IDCType::IDCZero);
+  TH2F* hAside2D = nullptr;
+  TH2F* hCside2D = nullptr;
```
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCCCDBHelper.cxx]:
```suggestion
```

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx [hunk 7]
```diff
@@ -99,6 +104,284 @@ unsigned int o2::tpc::IDCCCDBHelper<DataT>::getUngroupedIndexGlobal(const unsign
...
+    xBins[i] = xBins[nRowsIROC + 2 + nRowsOROC1] + (i - (nRowsIROC + 3 + nRowsOROC1) + 1) * 12;
+  }
+  xBins[nRowsIROC + 3 + nRowsOROC1 + nRowsOROC2] = xBins[nRowsIROC + 2 + nRowsOROC1 + nRowsOROC2] + gap2;
+
+  for (int i = nRowsIROC + 4 + nRowsOROC1 + nRowsOROC2; i < (nRowsIROC + 4 + nRowsOROC1 + nRowsOROC2 + nRowsOROC3); i++) {
+    xBins[i] = xBins[nRowsIROC + 3 + nRowsOROC1 + nRowsOROC2] + (i - (nRowsIROC + 4 + nRowsOROC1 + nRowsOROC2) + 1) * 15;
+  }
+
+  hAside2D = new TH2F("h_IDC0_radialProfile_Aside", "IDC0: Radial profile (A-Side)", 155, xBins, nbinsY, yMin, yMax);
+  hAside2D->GetXaxis()->SetTitle("x (mm)");
```
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCCCDBHelper.cxx]: Our default unit is `cm`, please adapt the code.

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx [hunk 8]
```diff
@@ -99,6 +104,284 @@ unsigned int o2::tpc::IDCCCDBHelper<DataT>::getUngroupedIndexGlobal(const unsign
...
+  hAside2D->GetYaxis()->SetTitle("IDC0 (ADC)");
+  hAside2D->GetZaxis()->SetTitle(zAxisTitle.data());
+  hAside2D->SetTitleOffset(1.05, "XY");
+  hAside2D->SetTitleSize(0.05, "XY");
+  hAside2D->SetStats(0);
+
+  hCside2D = new TH2F("h_IDC0_radialProfile_Cside", "IDC0: Radial profile (C-Side)", 155, xBins, nbinsY, yMin, yMax);
```
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCCCDBHelper.cxx]:
```suggestion
  auto hCside2D = new TH2F("h_IDC0_radialProfile_Cside", "IDC0: Radial profile (C-Side)", radialBinning.size()-1, radialBinning.data(), nbinsY, yMin, yMax);
```

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx [hunk 9]
```diff
@@ -99,6 +104,284 @@ unsigned int o2::tpc::IDCCCDBHelper<DataT>::getUngroupedIndexGlobal(const unsign
...
+  canv->cd(1);
+  hAside2D->Draw("colz");
+  hAside2D->ProfileX("profile_ASide", 1, -1, "d,same");
+  hAside2D->SetStats(0);
+
+  canv->cd(2);
+  hCside2D->Draw("colz");
+  hCside2D->ProfileX("profile_CSide", 1, -1, "d,same");
+  hAside2D->SetStats(0);
+
```
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCCCDBHelper.cxx:235]:
```suggestion

  hAside2D->SetBit(TObject::kCanDelete);
  hCside2D->SetBit(TObject::kCanDelete);
```

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx [hunk 10]
```diff
@@ -99,6 +104,284 @@ unsigned int o2::tpc::IDCCCDBHelper<DataT>::getUngroupedIndexGlobal(const unsign
...
+  }
+
+  canv->Divide(1, 2);
+  canv->cd(1);
+  hAside1D->Draw();
+  canv->cd(2);
+  hCside1D->Draw();
+
+  hAside1D->SetBit(TObject::kCanDelete);
+  hCside1D->SetBit(TObject::kCanDelete);
```
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCCCDBHelper.cxx:316]:
```suggestion
  canv->SetBit(TObject::kCanDelete);
```

File: Detectors/TPC/calibration/src/IDCDrawHelper.cxx [hunk 11]
```diff
@@ -108,6 +109,170 @@ void o2::tpc::IDCDrawHelper::drawSide(const IDCDraw& idc, const o2::tpc::Side si
...
+
+  return poly;
+}
+
+TH1F* o2::tpc::IDCDrawHelper::drawSide(const IDCDraw& idc, std::string_view type, const o2::tpc::Side side, int nbins1D, float xMin1D, float xMax1D)
+{
+  static const Mapper& mapper = Mapper::instance();
+  const int bufferSize = TH1::GetDefaultBufferSize();
+  TH1::SetDefaultBufferSize(Sector::MAXSECTOR * mapper.getPadsInSector());
+  std::string sideName = side == Side::A ? "A" : "C";
```
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCDrawHelper.cxx]: readability braces
```suggestion
  std::string sideName = (side == Side::A) ? "A" : "C";
```

File: Detectors/TPC/calibration/include/TPCCalibration/IDCDrawHelper.h [hunk 12]
```diff
@@ -56,6 +58,10 @@ class IDCDrawHelper
...
+  static TH1F* drawSide(const IDCDraw& idc, std::string_view type, const o2::tpc::Side side, const int nbins1D, const float xMin1D, const float xMax1D);
+  static void drawRadialProfile(const IDCDraw& idc, TH2F* hist, const o2::tpc::Side side, const int nbinsY, const float yMin, const float yMax);
```
Comment (wiechula) [inline Detectors/TPC/calibration/include/TPCCalibration/IDCDrawHelper.h]:
```suggestion
  static void drawRadialProfile(const IDCDraw& idc, TH2F& hist, const o2::tpc::Side side, const int nbinsY, const float yMin, const float yMax);
```

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx [hunk 13]
```diff
@@ -99,6 +104,284 @@ unsigned int o2::tpc::IDCCCDBHelper<DataT>::getUngroupedIndexGlobal(const unsign
...
+  hCside2D = new TH2F("h_IDC0_radialProfile_Cside", "IDC0: Radial profile (C-Side)", 155, xBins, nbinsY, yMin, yMax);
+  hCside2D->GetXaxis()->SetTitle("x (mm)");
+  hCside2D->GetYaxis()->SetTitle("IDC0 (ADC)");
+  hCside2D->GetZaxis()->SetTitle(zAxisTitle.data());
+  hCside2D->SetTitleOffset(1.05, "XY");
+  hCside2D->SetTitleSize(0.05, "XY");
+  hCside2D->SetStats(0);
+
+  IDCDrawHelper::drawRadialProfile(drawFun, hAside2D, Side::A, nbinsY, yMin, yMax);
+  IDCDrawHelper::drawRadialProfile(drawFun, hCside2D, Side::C, nbinsY, yMin, yMax);
```
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCCCDBHelper.cxx]:
```suggestion
  IDCDrawHelper::drawRadialProfile(drawFun, *hAside2D, Side::A, nbinsY, yMin, yMax);
  IDCDrawHelper::drawRadialProfile(drawFun, *hCside2D, Side::C, nbinsY, yMin, yMax);
```

File: Detectors/TPC/calibration/src/IDCDrawHelper.cxx [hunk 14]
```diff
@@ -108,6 +109,170 @@ void o2::tpc::IDCDrawHelper::drawSide(const IDCDraw& idc, const o2::tpc::Side si
...
+          h->Fill(idc.getIDC(sector, region, irow, padTmp));
+        }
+      }
+    }
+  }
+  TH1::SetDefaultBufferSize(bufferSize);
+  return h;
+}
+
+void o2::tpc::IDCDrawHelper::drawRadialProfile(const IDCDraw& idc, TH2F* hist, const o2::tpc::Side side, int nbinsY, float yMin, float yMax)
```
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCDrawHelper.cxx]:
```suggestion
void o2::tpc::IDCDrawHelper::drawRadialProfile(const IDCDraw& idc, TH2F& hist, const o2::tpc::Side side)
```

File: Detectors/TPC/calibration/src/IDCDrawHelper.cxx [hunk 15]
```diff
@@ -108,6 +109,170 @@ void o2::tpc::IDCDrawHelper::drawSide(const IDCDraw& idc, const o2::tpc::Side si
...
+            float tmpCoord = 848.5 + nRowsIROC * 7.5 + gap1 + nRowsOROC1 * 10 + gap2 + nRowsOROC2 * 12 + gap2 + (globRow - nRowsIROC - nRowsOROC1 - nRowsOROC2) * 15;
+            hist->Fill(tmpCoord, idc.getIDC(sector, region, irow, padTmp));
+          }
+        }
+      }
+    }
+  }
+}
+
+void o2::tpc::IDCDrawHelper::drawIDCZeroStackCanvas(const IDCDraw& idc, const o2::tpc::Side side, const std::string_view type, int nbins1D, float xMin1D, float xMax1D, TCanvas* outputCanvas, int integrationInterval)
```
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCDrawHelper.cxx]:
```suggestion
void o2::tpc::IDCDrawHelper::drawIDCZeroStackCanvas(const IDCDraw& idc, const o2::tpc::Side side, const std::string_view type, int nbins1D, float xMin1D, float xMax1D, TCanvas& outputCanvas, int integrationInterval)
```
And required changes below.

File: Detectors/TPC/calibration/include/TPCCalibration/IDCDrawHelper.h [hunk 16]
```diff
@@ -56,6 +58,10 @@ class IDCDrawHelper
...
+  static void drawIDCZeroStackCanvas(const IDCDraw& idc, const o2::tpc::Side side, const std::string_view type, const int nbins1D, const float xMin1D, const float xMax1D, TCanvas* outputCanvas, int integrationInterval);
```
Comment (wiechula) [inline Detectors/TPC/calibration/include/TPCCalibration/IDCDrawHelper.h]:
```suggestion
  static void drawIDCZeroStackCanvas(const IDCDraw& idc, const o2::tpc::Side side, const std::string_view type, const int nbins1D, const float xMin1D, const float xMax1D, TCanvas& outputCanvas, int integrationInterval);
```

File: Detectors/TPC/calibration/src/IDCDrawHelper.cxx [hunk 17]
```diff
@@ -108,6 +109,170 @@ void o2::tpc::IDCDrawHelper::drawSide(const IDCDraw& idc, const o2::tpc::Side si
   }
 }
 
+TH2Poly* o2::tpc::IDCDrawHelper::drawSide(const IDCDraw& idc, const o2::tpc::Side side, const std::string zAxisTitle)
```
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCDrawHelper.cxx:85]: call this function in the one above.

File: Detectors/TPC/calibration/src/IDCDrawHelper.cxx [hunk 18]
```diff
@@ -108,6 +109,170 @@ void o2::tpc::IDCDrawHelper::drawSide(const IDCDraw& idc, const o2::tpc::Side si
...
+            hist->Fill(tmpCoord, idc.getIDC(sector, region, irow, padTmp));
+          }
+          if (97 <= globRow && globRow < 127) {
+            float tmpCoord = 848.5 + nRowsIROC * 7.5 + gap1 + nRowsOROC1 * 10 + gap2 + (globRow - nRowsIROC - nRowsOROC1) * 12;
+            hist->Fill(tmpCoord, idc.getIDC(sector, region, irow, padTmp));
+          }
+          if (127 <= globRow && globRow < 152) {
+            float tmpCoord = 848.5 + nRowsIROC * 7.5 + gap1 + nRowsOROC1 * 10 + gap2 + nRowsOROC2 * 12 + gap2 + (globRow - nRowsIROC - nRowsOROC1 - nRowsOROC2) * 15;
+            hist->Fill(tmpCoord, idc.getIDC(sector, region, irow, padTmp));
+          }
```
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCDrawHelper.cxx]:
```suggestion
  const auto padNum = Mapper::getGlobalPadNumber(irow, ipad, region);
  const float padX = mapper.padCentre(padNum).x();
  hist.Fill(padX, idc.getIDC(sector, region, irow, padTmp));
```

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx [hunk 19]
```diff
@@ -99,6 +104,284 @@ unsigned int o2::tpc::IDCCCDBHelper<DataT>::getUngroupedIndexGlobal(const unsign
...
+  for (int i = nRowsIROC + 3 + nRowsOROC1; i < (nRowsIROC + 3 + nRowsOROC1 + nRowsOROC2); i++) {
+    xBins[i] = xBins[nRowsIROC + 2 + nRowsOROC1] + (i - (nRowsIROC + 3 + nRowsOROC1) + 1) * 12;
+  }
+  xBins[nRowsIROC + 3 + nRowsOROC1 + nRowsOROC2] = xBins[nRowsIROC + 2 + nRowsOROC1 + nRowsOROC2] + gap2;
+
+  for (int i = nRowsIROC + 4 + nRowsOROC1 + nRowsOROC2; i < (nRowsIROC + 4 + nRowsOROC1 + nRowsOROC2 + nRowsOROC3); i++) {
+    xBins[i] = xBins[nRowsIROC + 3 + nRowsOROC1 + nRowsOROC2] + (i - (nRowsIROC + 4 + nRowsOROC1 + nRowsOROC2) + 1) * 15;
+  }
+
+  hAside2D = new TH2F("h_IDC0_radialProfile_Aside", "IDC0: Radial profile (A-Side)", 155, xBins, nbinsY, yMin, yMax);
```
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCCCDBHelper.cxx]: I added a function for the binning to the Painter: https://github.com/AliceO2Group/AliceO2/pull/8772
Once it is merged you can use it:
```suggestion
  const auto radialBinning = painter::getRowBinningCM();
  
  auto hAside2D = new TH2F("h_IDC0_radialProfile_Aside", "IDC0: Radial profile (A-Side)", radialBinning.size()-1, radialBinning.data(), nbinsY, yMin, yMax);
```

File: Detectors/TPC/calibration/src/IDCDrawHelper.cxx [hunk 20]
```diff
@@ -108,6 +109,170 @@ void o2::tpc::IDCDrawHelper::drawSide(const IDCDraw& idc, const o2::tpc::Side si
...
+}
+
+void o2::tpc::IDCDrawHelper::drawRadialProfile(const IDCDraw& idc, TH2F* hist, const o2::tpc::Side side, int nbinsY, float yMin, float yMax)
+{
+  const int nRowsIROC = 63;
+  const int nRowsOROC1 = 34;
+  const int nRowsOROC2 = 30;
+  const int nRowsOROC3 = 25;
+  const float gap1 = 25.75;
+  const int gap2 = 21;
```
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCDrawHelper.cxx]:
```suggestion
const mapper& = Mapper::instance();
```

File: Detectors/TPC/calibration/src/IDCDrawHelper.cxx [hunk 21]
```diff
@@ -108,6 +109,170 @@ void o2::tpc::IDCDrawHelper::drawSide(const IDCDraw& idc, const o2::tpc::Side si
...
+          }
+          if (63 <= globRow && globRow < 97) {
+            hOROC1->Fill(idc.getIDC(sector, region, irow, padTmp));
+          }
+          if (97 <= globRow && globRow < 127) {
+            hOROC2->Fill(idc.getIDC(sector, region, irow, padTmp));
+          }
+          if (127 <= globRow && globRow < 152) {
+            hOROC3->Fill(idc.getIDC(sector, region, irow, padTmp));
+          }
```
Comment (wiechula) [inline Detectors/TPC/calibration/src/IDCDrawHelper.cxx]:
```suggestion
          if (region < 4) {
            hIROC->Fill(idc.getIDC(sector, region, irow, padTmp));
          } else if (region < 6) {
            hOROC1->Fill(idc.getIDC(sector, region, irow, padTmp));
          } else if (region < 8) {
            hOROC2->Fill(idc.getIDC(sector, region, irow, padTmp));
          } else {
            hOROC3->Fill(idc.getIDC(sector, region, irow, padTmp));
         }
```

**Review by wiechula — CHANGES_REQUESTED** [review]

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx (hunk 10 above)
Comment (tklemenz) [reply Detectors/TPC/calibration/src/IDCCCDBHelper.cxx:316]: I think this should stay as it is.

**Comment by tklemenz** [comment]
I implemented everything except the binning, where I wait for the PR in the painter to be merged. 
@wiechula  Also please check the one comment above.

File: Detectors/TPC/calibration/src/IDCCCDBHelper.cxx (hunk 10 above)
Comment (wiechula) [reply Detectors/TPC/calibration/src/IDCCCDBHelper.cxx:316]: Yes, sure. My bad!

**Review by wiechula — CHANGES_REQUESTED** [review]
https://github.com/AliceO2Group/AliceO2/pull/8772 was merged, please change your code as suggested.

**Review by wiechula — APPROVED** [review]

**Comment by tklemenz** [comment]
Should I leave WIP until the PR it depends on is merged?

**Comment by wiechula** [comment]
Isn't it already merged?

**Comment by tklemenz** [comment]
Ah, my bad. I looked at the wrong PR. So I remove it.

**Comment by wiechula** [comment]
@shahor02 , @davidrohr , can this be merged?

**Comment by wiechula** [comment]
can this be merged?

**Pull request merged**